packages/protocol/scripts/publish.ts

```diff
@@ -0,0 +1,57 @@
+/* tslint:disable no-console */
+import FormData from 'form-data'
+import fs from 'fs'
+import fetch from 'node-fetch'
+import path from 'path'
+
+const sleep = (ms) => {
+  return new Promise((resolve) => setTimeout(resolve, ms))
+}
+
+async function main() {
+  const artifactBasePath = './build/contracts'
+  const artifactPaths = fs.readdirSync(artifactBasePath)
+  const network = process.env.NETWORK || 'development'
+
+  console.log('Uploading sources & metadata')
+  console.log('============================')
+  console.log(artifactPaths)
+
+  for (const _path of artifactPaths) {
+    const artifact = require(path.join(process.cwd(), artifactBasePath, _path))
+
+    console.log()
+    console.log(artifact.contractName)
+    console.log('-'.repeat(artifact.contractName.length))
+
+    const chainId = Object.keys(artifact.networks)[0]
```

Comment:
Why are we not going to get access to the deployment artifacts in the moment that we are making a new deployment?

Reply:
Yeah, so I think my point is that it'd be best if we decouple this process from the moment we make a new deployment because we do not persist deployment artifacts. So even if the typical process would be to run this immediately after a deployment, we want to still be able to recover from the situation where we forgot to run this/this script crashed/etc.